app/javascript/packs/base.jsx

```diff
@@ -144,3 +144,25 @@ getInstantClick().then((spa) => {
 });
 
 initializeNav();
+
+async function loadCreatorSettings() {
```

Comment:
This pack file is loaded even in the creator settings page. I thought of creating a separate one, but it means adding another conditional in how we load pack files. With this approach, it's a quick check on the route and then the controller is dynamically loaded.

Reply:
ok, actually this seems to be how we load the stimulus controller to the user facing side.

Reply:
Do you have any long term plans for this, if we wanted to load another stimulus controller on the user facing side, would we generalize this function or would there be a new async loadXYZ for each controller?

Reply:
> Do we have any long term plans for this - it seems to be the first time we use a stimulus controller in a non-admin layout view.  
> 
> 
> 
> How does it scale? If we wanted to load another stimulus controller here, would we generalize this function or would there be a new async loadXYZ for each controller? 
> 
> 
> 
> 
> 
> Also, if we wanted to use the same stimulus controller on the admin, would we be able to do that? I can see this being the case for image upload, we'd use the same controller from creator settings on the admin for for image upload.

It's being loaded dynamically so it does not affect page loads of the client-facing portion of the application aside from a quick Bboolean check on the url.

As you mention, it is the first time we do this. We could probably take a page out of the rails handbook and use the route to determine which controller to load on the client-side. That way we could have one function to dynamically load a Stimulus controller if one exists. Something along those lines. We can discuss it further as I'm 100% open to suggestions/I haven't thought about it too much yet. 😎